### ISSUE TITLE
Help needed with dataset conversions!

### ISSUE THREAD

**Issue by ujwalakoriraj** [Paulymorphous/skeyenet#13]
I have downloaded the dataset from the website given in the repo. I can neither convert all the .tiff dataset (after splitting for train and test) into numpy-array, to go ahead and compile the model, nor create h5py files for images and corresponding masks (before splitting for train and test) to load into the train_test_split() function. I have been encountering errors in doing so. Have got problem loading the .tiff also.
Tried every possible way!
Please help!

This is the code to convert the entire .tiff dataset into numpy-array:
********
from glob import glob
import PIL
import os
import numpy as np
import matplotlib.pyplot as plt

# train_images='D:/python/building_detection/8_building/data/building/train/images/samples'
# train_masks='D:/python/building_detection/8_building/data/building/train/masks/samples'
# test_images='D:/python/building_detection/8_building/data/building/test/images/samples'
# test_masks='D:/python/building_detection/8_building/data/building/test/masks/samples'

# images1 = glob(os.path.join(train_images, "*.tiff"))
# images1 = glob(os.path.join(train_masks, "*.tiff"))
# images3 = glob(os.path.join(test_images, "*.tiff"))
# images4 = glob(os.path.join(test_masks, "*.tiff"))

#for load in images1:
image=PIL.Image.open('D:/python/building_detection/8_building/data/building/train/images/samples/1_24029050_15.tiff')
image_array=np.array(image)
    
# print(image_array.shape)   

*************

Got the following errors:
*********
(a never-ending...)
TIFFReadDirectory: Failed to read directory at offset 1706.
tempfile.tif: Not a TIFF or MDI file, bad magic number 128 (0x80).
tempfile.tif: Not a TIFF or MDI file, bad magic number 33285 (0x8205).
tempfile.tif: Not a TIFF or MDI file, bad magic number 3349 (0xd15).
TIFFFetchDirectory: Sanity check on directory count failed, this is probably not a valid IFD offset.
TIFFReadDirectory: Failed to read directory at offset 2758.
tempfile.tif: Not a TIFF or MDI file, bad magic number 16256 (0x3f80).
tempfile.tif: Not a TIFF or MDI file, bad magic number 33797 (0x8405).
tempfile.tif: Not a TIFF or MDI file, bad magic number 3333 (0xd05).
TIFFFetchDirectory: Can not read TIFF directory count.
TIFFReadDirectory: Failed to read directory at offset 194470.
tempfile.tif: Not a TIFF or MDI file, bad magic number 128 (0x80).
tempfile.tif: Not a TIFF or MDI file, bad magic number 33031 (0x8107).
tempfile.tif: Not a TIFF or MDI file, bad magic number 65308 (0xff1c).
TIFFFetchDirectory: Can not read TIFF directory count.
TIFFReadDirectory: Failed to read directory at offset 194470.
*********

Code for h5py creation (the file gets created, but I have problem accessing the images later. A blank pair of normal-parenthesis shows for print(images.shape)):

************
import h5py
import os
import PIL
from glob import glob
import numpy as np

images_path="D:/python/building_detection/8_building/data/building/targets"

#images_path="D:/python/building_detection/8_building/data/building/targets"

#list=os.listdir(images_path)
images = glob(os.path.join(images_path, "*.tif"))

IMAGE_HEIGHT = IMAGE_WIDTH = 256
NUM_CHANNELS = 3

with h5py.File('testing.h5', 'w') as hf:
    for i,img in enumerate(images): 
        image=PIL.Image.open(img)
        Xset = hf.create_dataset(
                #name='X'+str(i),
                name='all_images',
                
                data=image,
                )
************

**Comment by jalotra** [comment]
Was this solved ?

**Comment by ujwalakoriraj** [comment]
Yes, I used a different dataset which contained images in .png format.

**Comment by Sneh09** [comment]
> Yes, I used a different dataset which contained images in .png format.

Could you please share how you did it. As I have input in and masks in tiff format
